calicoctl/calico_ctl/pool.py

```diff
@@ -122,19 +122,61 @@ def ip_pool_add(cidrs, version, ipip, masquerade):
         print "IP in IP not supported for IPv6 pools"
         sys.exit(1)
 
-    # TODO Reject any cidrs that overlap with existing cidrs in the pool
-    pools = []
+    current_pool_nets = [pool.cidr for pool in client.get_ip_pools(version)]
+    new_pools = []
+
+    # Validate pool CIDRs
     for cidr in cidrs:
+        cidr_net = IPNetwork(cidr)
```

Comment:
`cidr` and `cidr_net` are still being thrown around interchangeably, which is making things harder to follow. I tested and confirmed this to be true:
```
str(IPNetwork(cidr)) == cidr
```

This means that your log messages can `%s` in IPNetwork instances and get the same effect as if they were fed string representation instead.

Therefore, I think you should update this line to:
`cidr = IPNetwork(cidr)`

From scanning over the code, you will not need the string verion of the cidr at all. Additionally, when you construct an IPPool using the cidr on line 148, the init method for IPPool can handle both strings and IPNetwork objects, so it to will handle the IPNetwork object just fine